interpreter/src/macros.rs

```diff
@@ -1,12 +1,25 @@
 #[macro_export]
 macro_rules! impl_conversions {
     // Capture pairs separated by commas, where each pair is separated by =>
-    ($($target_type:ty => $value_variant:path),* $(,)?) => {
+    ($($target_type:ty $(as $cast:ty)? => $value_variant:path),* $(,)?) => {
         $(
             impl FromValue for $target_type {
                 fn from_value(expr: &Value) -> Result<Self, ExecutionError> {
                     if let $value_variant(v) = expr {
-                        Ok(v.clone())
+                        $(if <$target_type>::MAX as $cast < *v {
+                            return Err(ExecutionError::CastOverflow {
+                                value: *v as f64,
+                                source_ty: std::any::type_name::<$cast>(),
+                                target_ty: std::any::type_name::<$target_type>(),
+                            })
+                        } else if <$target_type>::MIN as $cast > *v {
+                            return Err(ExecutionError::CastOverflow {
+                                value: *v as f64,
+                                source_ty: std::any::type_name::<$cast>(),
+                                target_ty: std::any::type_name::<$target_type>(),
+                            })
+                        })?
+                        Ok(v.clone() $(as $cast as $target_type)?)
```

Comment:
This change allows conversion without indirection over `serde` which means that serde can be fully disabled now and the crate will still work with all basic number types.

Double casting in `$(as $cast as $target_type)?` is needed so macro_rules can figure out the cast should be only added when `as $cast` part is specified.